### DISCUSSION TITLE
mass updates

### DISCUSSION THREAD

**Discussion by ChrisSantiago82** [laravel/framework#45657] [Q&A]
I have a question, maybe there is an easy answer and I can just not find it.

In eloquent with have the upsert method. To mass  create or update entries.

Is there something similar just for mass updates. Like I have an unique field, and I have one other field I want to update.
Instead of looping and make hundreds or thousands or queries.

**Comment by henzeb** [comment]
Does this what you want?
https://laravel.com/docs/9.x/eloquent#mass-updates

**Reply by ChrisSantiago82** [reply]
Not really. Like in this example the value of all is set to 1.

Flight::where('active', 1)
      ->where('destination', 'San Diego')
      ->update(['delayed' => 1]);

But how what I'm looking for is something like this, just with update and without insert:

Flight::upsert([
    ['departure' => 'Oakland', 'destination' => 'San Diego', 'price' => 99],
    ['departure' => 'Chicago', 'destination' => 'New York', 'price' => 150]
], ['departure', 'destination'], ['price']);

**Reply by ChrisSantiago82** [reply]
Let's say you have to update the price of a hundred flights, but each flight has a different price
